prow/plugins/updateconfig/updateconfig.go

```diff
@@ -117,19 +122,30 @@ func Update(fg FileGetter, kc corev1.ConfigMapInterface, name, namespace string,
 			return fmt.Errorf("get file err: %v", err)
 		}
 		logger.WithFields(logrus.Fields{"key": upd.Key, "filename": upd.Filename}).Debug("Populating key.")
-		value := string(content)
+		value := content
 		if upd.GZIP {
 			buff := bytes.NewBuffer([]byte{})
 			// TODO: this error is wildly unlikely for anything that
 			// would actually fit in a configmap, we could just as well return
 			// the error instead of falling back to the raw content
-			if _, err := gzip.NewWriter(buff).Write(content); err != nil {
+			z := gzip.NewWriter(buff)
+			if _, err := z.Write(content); err != nil {
 				logger.WithError(err).Error("failed to gzip content, falling back to raw")
 			} else {
-				value = buff.String()
+				if err := z.Close(); err != nil {
+					logger.WithError(err).Error("failed to flush gzipped content (!?), falling back to raw")
+				} else {
```

Comment:
Shouldn't we just log the error closing and always do this step?

Reply:
No, because if we don't close the content of `buff` is nonsense (which is the gzip bug I mention in my above github comment).